airbyte-integrations/connectors/source-datascope/source_datascope/source.py

```diff
@@ -0,0 +1,160 @@
+            stream_state: Mapping[str, Any],
+            stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping]:
+        return response.json()
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, any]:
+        # This method is called once for each record returned from the API to compare the cursor field value in that record with the current state
+        # we then return an updated state object. If this is the first time we run a sync or no state was passed, current_stream_state will be None.
+        if current_stream_state is not None and 'updated_at' in current_stream_state:
+            current_parsed_date = datetime.strptime(current_stream_state['updated_at'], '%Y-%m-%dT%H:%M:%S%z')
+            last_date = latest_record['updated_at'].split('.')[0] + '+0000'
+            if self.schema_type == 'dynamic':
+                latest_record_date = datetime.strptime(last_date, '%d/%m/%Y %H:%M%z')
+            else:
+                latest_record_date = datetime.strptime(last_date, '%Y-%m-%dT%H:%M:%S%z')
+            return {'updated_at': max(current_parsed_date, latest_record_date).strftime('%Y-%m-%dT%H:%M:%S%z')}
+        else:
+            return {'updated_at': self.start_date.strftime('%Y-%m-%dT%H:%M:%S%z')}
+
+    def _chunk_date_range(self, start_date: datetime) -> List[Mapping[str, any]]:
+        """
+        Returns a list of each day between the start date and now.
+        The return value is a list of dicts {'date': date_string}.
+        """
+        dates = []
+        while start_date < datetime.now(start_date.tzinfo):
+            dates.append({'updated_at': start_date.strftime('%Y-%m-%dT%H:%M:%S%z')})
+            start_date += timedelta(days=7)
+        return dates
+    def stream_slices(self, sync_mode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None) -> Iterable[
+        Optional[Mapping[str, any]]]:
+        start_date = datetime.strptime(stream_state['updated_at'], '%Y-%m-%dT%H:%M:%S%z') if stream_state and 'updated_at' in stream_state else self.start_date
+        return self._chunk_date_range(start_date)
+
+
+
+
+class Forms(DatascopeIncrementalStream):
+    primary_key = "form_answer_id"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        if self.schema_type == 'dynamic':
+            return "v2/airbyte"
+        else:
+            return "answers"
+
+# Source
+class SourceDatascope(AbstractSource):
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        return True, None
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        auth = NoAuth()
```

Comment:
can use `TokenAuthenticator` instead of using headers directly

Reply:
How I can use that to then set the self.token attribute??

Reply:
when using it, you don't need to set `self.token` -- the authenticator will set it on the headers directly. See [TokenAuthenticator class](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/http/auth/token.py) and how authenticator is used in [HttpStream](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py#L235) class

Reply:
Great, I understand. Going to change that